commit b50d2a8c86bd5b2b0d6cd3926114c80862bc4c85
Author: Philipp Hortmann <philipp.g.hortmann@gmail.com>
Date:   Sun Aug 27 15:27:04 2023 +0200

    staging: rtl8192e: Remove unsupported mode IW_MODE_ADHOC
    
    Remove unsupported mode IW_MODE_ADHOC because this mode is only partially
    coded in the driver.
    
    rtllib_rx_mgt() can only handle probe requests in IW_MODE_ADHOC and no
    authentication or association frames.
    
    Signed-off-by: Philipp Hortmann <philipp.g.hortmann@gmail.com>
    Link: https://lore.kernel.org/r/20230827132704.GA12506@matrix-ESPRIMO-P710
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/staging/rtl8192e/rtllib_softmac_wx.c b/drivers/staging/rtl8192e/rtllib_softmac_wx.c
index 0b690f0ffeef..48703d93df30 100644
--- a/drivers/staging/rtl8192e/rtllib_softmac_wx.c
+++ b/drivers/staging/rtl8192e/rtllib_softmac_wx.c
@@ -258,7 +258,6 @@ int rtllib_wx_set_mode(struct rtllib_device *ieee, struct iw_request_info *a,
 	mutex_lock(&ieee->wx_mutex);
 	switch (wrqu->mode) {
 	case IW_MODE_MONITOR:
-	case IW_MODE_ADHOC:
 	case IW_MODE_INFRA:
 		break;
 	case IW_MODE_AUTO: